### PR TITLE
Earlgrey/create folder ui tests

### PR DESCRIPTION
**Pull request by javiergonzper**
## Description
This PR add 6 UI tests about the view of "Create folder"

## Related Issue
https://github.com/owncloud/ios-app/issues/58

## Motivation and Context
This PR add more tests to UI

## How Has This Been Tested?
Net necessary Bitrise will tests it by us

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: ownCloudTests/File List/CreateFolderTests.swift [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+	func testCreateFolder() {
+
+		if let bookmark: OCBookmark = UtilsTests.getBookmark() {
+
+			let folderName = "New Folder"
+
+			//Mocks
+			self.mockOCoreForBookmark(mockBookmark: bookmark)
+			self.mockQueryPropfindResults(resourceName: "PropfindResponse", basePath: "/remote.php/dav/files/admin", state: .contentsFromCache)
+			self.showFileList(bookmark: bookmark, issue: nil)
```
Comment (pablocarmu) [inline ownCloudTests/File List/CreateFolderTests.swift]: I've seen that you're calling `self.showFileList(bookmark: bookmark, issue: nil)` with the issue being nil multiple times. Maybe you want to change the declaration of the method to something like:

```Swift
showFileList(bookmark: OCBookmark, issue: OCIssue? = nil)
```

File: ownCloudTests/File List/Subclasses/MockClientRootViewController.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+import UIKit
+import ownCloudSDK
+@testable import ownCloud
+
+class MockClientRootViewController: ClientRootViewController {
+
+	var query:MockOCQuery!
```
Comment (pablocarmu) [inline ownCloudTests/File List/Subclasses/MockClientRootViewController.swift]: You don't need here the `!`, you can code this like: 
```Swift

private var query: MockOCQuery
private var mockedCore: MockOCCore

convenience init(core: MockOCCore, query: OCQuery, bookmark: OCBookmark) {
 self.query = query
 self.mockedCore = core
 self.mockedCore.delegate = self
 self.init(bookmark: bookmark)
}
```

File: ownCloudTests/File List/Subclasses/MockOCCore.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+//
+//  Created by Javier Gonzalez on 10/01/2019.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+import ownCloudSDK
+
+class MockOCCore: OCCore {
+
+	var query:MockOCQuery?
```
Comment (pablocarmu) [inline ownCloudTests/File List/Subclasses/MockOCCore.swift]: If you don't set `query` from outside with a `nil` value, you don't need this to be optional, the code would look like:
```Swift

        var query: MockOCQuery

	convenience init(query: MockOCQuery, bookmark: OCBookmark, issue: OCIssue?) {
                self.query = query
		self.init(bookmark: bookmark)
		self.issue = issue
	}
```

File: ownCloudTests/Tools/UtilsTests.swift [hunk 4]
```diff
@@ -22,7 +22,7 @@ class UtilsTests {
 
 	static func getBookmark(authenticationMethod: OCAuthenticationMethodIdentifier = OCAuthenticationMethodIdentifier.basicAuth, bookmarkName: String = "Server name") -> OCBookmark? {
 
-		let mockUrlServer: String = "https://mock.owncloud.com/"
+		let mockUrlServer: String = "https://demo.owncloud.com/"
```
Comment (pablocarmu) [inline ownCloudTests/Tools/UtilsTests.swift]: Take a look at this change

File: ownCloudTests/File List/CreateFolderTests.swift (hunk 1 above)
Comment (javiergonzper) [reply ownCloudTests/File List/CreateFolderTests.swift]: Yes, most of the times is nil, thank you!

File: ownCloudTests/File List/CreateFolderTests.swift [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+		if let bookmark: OCBookmark = UtilsTests.getBookmark() {
+			//Mocks
+			self.mockOCoreForBookmark(mockBookmark: bookmark)
+			self.mockQueryPropfindResults(resourceName: "PropfindResponse", basePath: "/remote.php/dav/files/admin", state: .contentsFromCache)
+			self.showFileList(bookmark: bookmark)
+
+			//Actions
+			EarlGrey.select(elementWithMatcher: grey_accessibilityID("sort-bar.leftButton")).perform(grey_tap())
+
+			//Assets
```
Comment (jesmrec) [inline ownCloudTests/File List/CreateFolderTests.swift]: Assets -> Asserts

File: ownCloud/Client/ClientQueryViewController.swift [hunk 6]
```diff
@@ -275,6 +275,9 @@ class ClientQueryViewController: UITableViewController, Themeable {
 		let cell = tableView.dequeueReusableCell(withIdentifier: "itemCell", for: indexPath) as? ClientItemCell
 		let newItem = itemAtIndexPath(indexPath)
 
+		cell?.accessibilityIdentifier = newItem.name
+		print(newItem.name)
```
Comment (hosy) [inline ownCloud/Client/ClientQueryViewController.swift]: print() should be removed

File: ownCloudTests/File List/CreateFolderTests.swift (hunk 5 above)
Comment (javiergonzper) [reply ownCloudTests/File List/CreateFolderTests.swift]: Changed here and in other tests that were wrong 👍

**Pull request merged**